kratos/containers/container_expression/expressions/expression.h

```diff
@@ -37,14 +37,87 @@ class KRATOS_API(KRATOS_CORE) Expression {
     ///@name Type definitions
     ///@{
 
-    using Pointer = Kratos::intrusive_ptr<Expression>;
+    using Pointer = Kratos::intrusive_ptr<const Expression>;
 
     using IndexType = std::size_t;
```

Comment:
If you want `Expression` to act as a proper container, I suggest adding
- member alias `value_type` (the "stored" type - same as `value_type` of `iterator`)
- member alias `size_type` (return type of `Expression::size()`)
- member alias `const_iterator` (and `iterator` if you allow mutable access)
- member function `size` (number of components in the container)

Reply:
My laziness strikes again....

It is a pain to have iterators :/

Reply:
done :)